### ISSUE TITLE
Allow authentication via RADIUS (or AD) and Certificate

### ISSUE THREAD

**Issue by RolfWojtech** [SoftEtherVPN/SoftEtherVPN#226]
Is there a way to combine Radius or AD authentication with a certificate requirement?
From a security perspective it seems sensible to attach a certificate to the machine to prevent attackers who stole username and password from connecting. The additional Radius or AD request would ensure that you can easily disable accounts when someone leaves the company or group.

I'm open to dirty hacks and workarounds, any ideas welcome.

**Comment by tlchan** [comment]
You already can set the expiration date for the account, just set it for a past date to disable the account.

**Comment by moatazelmasry2** [comment]
@RolfWojtech Yes I think this is a nice feature that would tighten security of SE

**Comment by elliot-labs** [comment]
MFA would be ideal.
This is a must have.
We are in an age of stolen credentials and need more than one Auth method.

**Comment by chipitsine** [comment]
I'm going to collect "enterprise" tasks together. Thank for the idea

**Comment by chipitsine** [comment]
as a side note (I was running  pretty big vpn installation for 5+ years), I have a question - for example, you deployed MSI using GPO. ok. How to deal with "road warriors" ? i.e. people who use own devices to connect to vpn ? no GPO, nothing. 

from that point of view, GPO is not very usefull (but I would to hear that I'm wrong)
